P0009/P0009r11.md

```diff
@@ -810,36 +835,40 @@ Y8b.     .d8""8b. Y88b. Y8b.     888  888 Y88b.       X88
 ```cpp
 namespace std {
 
-template<ptrdiff_t... Extents>
+template<size_t... Extents>
 class extents {
 public:
-  using index_type = ptrdiff_t;
+  using size_type = size_t;
 
   // [mdspan.extents.cons], Constructors and assignment
   constexpr extents() noexcept = default;
   constexpr extents(const extents&) noexcept = default;
   constexpr extents& operator=(const extents&) noexcept = default;
 
-  template<ptrdiff_t... OtherExtents>
+  template<size_t... OtherExtents>
     constexpr extents(const extents<OtherExtents...>&) noexcept;
-  template<class... IndexType>
-    constexpr extents(IndexType...) noexcept;
-  template<class IndexType>
-    constexpr extents(const array<IndexType, rank_dynamic()>&) noexcept;
-  template<ptrdiff_t... OtherExtents>
+  template<class... SizeTypes>
+    constexpr extents(SizeTypes...) noexcept;
+  template<class SizeType>
+    constexpr extents(const array<SizeType, rank_dynamic()>&) noexcept;
+  template<size_t... OtherExtents>
     constexpr extents& operator=(const extents<OtherExtents...>&) noexcept;
 
   // [mdspan.extents.obs], Observers of the domain multidimensional index space
   static constexpr size_t rank() noexcept { return sizeof...(Extents); }
   static constexpr size_t rank_dynamic() noexcept 
-    { return (size_t(Extents == dynamic_extent) + ...); }
-  static constexpr index_type static_extent(size_t) noexcept
+    { return ((Extents == dynamic_extent) + ...); }
+  static constexpr size_type static_extent(size_t) noexcept
     { return rank() - rank_dynamic(); } 
```

Comment:
this is weird. Shouldn't static_extent(k) return dynamic_extent if that guy is not static, and extent(k) if it is?